doc/conf.py

```diff
@@ -103,8 +111,6 @@
 
 todo_include_todos = False
 
-numfig = True
```

Comment:
if numfig increases build time, let's just drop its usage and so the option. I think it's used in just a few locations. Regarding the extensions:

- kconfig: we should introduce tracking of origin Kconfig files, and to mtime comparison on incremental builds. Not immediate as kconfiglib doesn't expose this AFAIK.
- breathe: it's always been slow and clunky, I doubt it will ever be _fixed_. IMO, we should have moved to Doxygen alone a long time ago, Sphinx+breathe doesn't scale to big projects like Zephyr. FYI a proposal I made a long time ago https://github.com/zephyrproject-rtos/zephyr/pull/39702

minimal build should also disable DT bindings docs, so we have a single way of building "fast" docs.

Reply:
+1 on getting rid of numfig. In fact, IIRC there are more places where people have manually numbered and referenced their figures than there are occurrences of numfig being used. I would be happy to implement the change.

Reply:
In my "benchmarking" I found `numfig` to be the most guilty by far. @kartben if there is consensus (please thumb up or down @kartben's comment) then it would be awesome if you can get rid of it entirely.

Reply:
> breathe: it's always been slow and clunky, I doubt it will ever be fixed.

I once tried to debug an incremental build performance issue in breathe: https://github.com/breathe-doc/breathe/issues/420

I had to give up and I'll never try again. I found the codebase to be one of the most object-obfuscated I ever tried to make sense of - without the corresponding C++ speed.

Reply:
So @kartben you got 3 thumbs up and 0 thumbs down. I would hate you doing and submitting the `numfig` cleanup and get rejected but... go?